influxdb_plugin.py

```diff
@@ -101,76 +102,102 @@ def process(cluster, stats):
             # if it doesn't convert to a different type then it is a string
             # value, which does not really make sense for InfluxDB, but oh well.
             pass
-        _process_stat(stat.key, stat.time, stat.value, tags, influxdb_points)
-        num_points = len(influxdb_points)
+        influxdb_point = \
+                _influxdb_point_from_stat(
+                        stat.time, tags, stat.key, stat.value)
+        if influxdb_point is not None and len(influxdb_point["fields"]) > 0:
+            influxdb_points.append(influxdb_point)
+            num_points += 1
         if num_points > MAX_POINTS_PER_WRITE:
             points_written += _write_points(influxdb_points, num_points)
             influxdb_points = []
+            num_points = 0
     # send left over points to influxdb
     num_points = len(influxdb_points)
     if num_points > 0:
         points_written += _write_points(influxdb_points, num_points)
     LOG.debug("Done processing stats, wrote %d points.", points_written)
 
 
-def _process_stat_dict(stat_key, stat_time, stat_value, tags, influxdb_points):
+def _add_field(fields, field_name, field_value, field_value_type):
+    if field_value_type == long and (field_value >> 63) >= 1:
+        # influxdb max is 64 bit signed integer, so bit shifting by 63 bits to
+        # identify if the 64th or greater bit is set and if so then convert to
+        # a float (which unfortunately means that we'll be throwing away some
+        # bits, although its the least significant bits, so not that bad).
+        field_value = float(field_value)
```

Comment:
Yea, you are right. I'm starting to come around to the idea of just converting everything to float64. I really can't think of any downside other than it seems wrong in principle. And the upside of it preventing things from breaking and you having to either convert your existing measurements to float or start from scratch is a pretty strong one.